browser/src/communication/federated/federated_client.ts

```diff
@@ -119,13 +123,34 @@ export class FederatedClient extends Client {
     }
   }
 
+  private addDifferentialPrivacy (currentRoundWeights: Weights): Weights {
+    const weightsDiff = List(currentRoundWeights)
+      .zip(List(this.previousRoundWeights))
+      .map(([w1, w2]) => w1.add(-w2))
+
+    const normBound = weightsDiff.map((w) => w.norm().dataSync()[0]).max()
+
+    const noisyWeightsDiff = weightsDiff.map((w) => {
+      const clipped = w.div(Math.max(1, w.norm().dataSync()[0] / normBound))
```

Comment:
the clipping radius (the maximum norm the grad can be after clipping) should be a parameter in the task config as well. if set to infty, there will be no clipping